hnn_core/viz.py

```diff
@@ -554,7 +573,11 @@ def plot_spikes_raster(cell_response, trial_idx=None, ax=None, show=True):
     ax.set_facecolor('k')
     ax.set_xlabel('Time (ms)')
     ax.get_yaxis().set_visible(False)
-    ax.set_xlim(left=0)
+
+    if len(cell_response.times) > 0:
```

Comment:
this is great @katduecker!

does `plot_spikes_histogram()` need the same logic?

Reply:
haven't touched plot_spikes_hist(), but it seems that xlim is not set explicitly in this function and tmin and tmax are not required in that function.

Reply:
@ntolley this turned out to be more complicated than anticipated. For some reason, I cannot set xlim outside plot_spikes_histogram() (can show you in person). Not sure why that is because it works if I change xlim inside the function, after the histogram loop. It seems that in this case, we might actually want to add tmin and tmax to plot_spikes_histogram() ... I wonder if it would be more user-friendly in this case to just keep tmin and tmax for all plotting functions?